geoengine_geoname_geocoder/res_partner.py

```diff
@@ -94,31 +106,28 @@ def geocode_from_geonames(self, cursor, uid, ids, srid='900913', strict=True, co
                 try:
                     url = base_url + urlencode(filters)
                     answer = urlopen(url)
-                    data = {'geo_point': self._get_point_from_reply(answer)}
-                    add.write(data)
-                    # We use postgres to do projection in order not to install GDAL dependences
-                    sql = ("UPDATE res_partner"
-                           "  SET geo_point = ST_Transform(st_SetSRID(geo_point, 4326), %s)"
-                           "  WHERE id = %s")
-                    cursor.execute(sql, (srid, add.id))
+                    add.geo_point = self._get_point_from_reply(answer)
                 except Exception, exc:
                     logger.exception('error while updating geocodes')
                     if strict:
-                        raise except_osv(_('Geoencoding fails'), str(exc))
-        return ids
-
-    def write(self, cursor, uid, ids, vals, context=None):
-        res = super(ResPartner, self).write(cursor, uid, ids, vals, context=None)
-        do_geocode = self._can_geocode(cursor, uid, context)
-        if do_geocode and "country_id" in vals or 'city' in vals or 'zip' in vals:
-            self.geocode_from_geonames(cursor, uid, ids, context=context)
+                        raise except_orm(_('Geoencoding fails'), str(exc))
+
+    @api.multi
+    def write(self, vals):
+        res = super(ResPartner, self).write(vals)
+        do_geocode = self._can_geocode()
+        if do_geocode \
+            and "country_id" in vals \
+            or 'city' in vals \
+                or 'zip' in vals:
```

Comment:
About logical operators needing to be at the end of a line.

Maybe the `\` makes it so pep8 doesn't register it.

Anyway, isn't it simpler to use `issubset`?